libsolidity/interface/CompilerStack.cpp

```diff
@@ -1413,7 +1421,7 @@ void CompilerStack::compileContract(
 
 	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
 
-	std::shared_ptr<Compiler> compiler = std::make_shared<Compiler>(m_evmVersion, m_revertStrings, m_optimiserSettings);
+	std::shared_ptr<Compiler> compiler = std::make_shared<Compiler>(m_evmVersion, std::nullopt, m_revertStrings, m_optimiserSettings);
```

Comment:
I actually didn't mean just passing ``std::nullopt`` here, but not adding the eof version option to ``Compiler`` (and ``CompilerContext``) in the first place (and just have them pass ``std::nullopt`` to the ``Assembly``s they need and such)
And an assert
```
solUnimplementedAssert(!m_eofVersion, "EOF is not supported for legacy code generation");
```
here also wouldn't hurt, together with the ``!m_viaIR`` assert below maybe.